File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        int lt = 0, rt = 0;
        
        for(int i = 0; i < goal.length; i++){
            if(lt < cards1.length && goal[i].equals(cards1[lt])){
                lt++;
            }else if (rt < cards2.length && goal[i].equals(cards2[rt])){
                rt++;
            }else{
                answer="No";
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    var answer = 'Yes';
    let lt = 0;
    let rt = 0;
    
    for(let i = 0; i < goal.length; i++){
        if(cards1[lt] === goal[i]){
            lt++;
        }else if(cards2[rt] === goal[i]){
            rt++;
        }else{
            return 'No';
        }
    }
    return answer;
}